app/src/main/java/com/github/sdpteam15/polyevents/fragments/MapsFragment.kt

```diff
@@ -107,6 +130,90 @@ class MapsFragment : Fragment(), OnMapReadyCallback, OnPolylineClickListener,
         GoogleMapHelper.interactionMarker(p0)
     }
 
+    override fun onMyLocationButtonClick(): Boolean {
+        // Return false to indicate that we did not consume the event. So the default behavior
+        // still occurs (which is to move the camera to the current location.
+        return false
+    }
+
+    /**
+     * Activate "my location" tracking.
+     * (source : https://developers.google.com/maps/documentation/android-sdk/location#kotlin)
+     */
+    private fun activateMyLocation() {
+        if (context?.let {
+                ContextCompat.checkSelfPermission(
+                    it,
+                    Manifest.permission.ACCESS_FINE_LOCATION
+                )
+            }
+            == PackageManager.PERMISSION_GRANTED) {
+            GoogleMapHelper.map!!.isMyLocationEnabled = true
+
+            // Hide the built-in location button (but DO NOT disable it !)
+            getBuiltInLocationButton().isVisible = false
+
+            // Change the appearance of the location button.
+            setLocationIcon(true)
+            useUserLocation = true
+        }
+    }
+
+    /**
+     * Update the location button icon according to whether the location is on or not
+     * @param locationIsOn : boolean, true if the location is currently used
+     */
+    private fun setLocationIcon(locationIsOn: Boolean) {
+        val idOfResource: Int = if (locationIsOn) {
+            R.drawable.ic_location_on
+        } else {
+            R.drawable.ic_location_off
+        }
+        requireView().findViewById<FloatingActionButton>(R.id.id_location_button)
+            .setImageResource(idOfResource)
+        locationButton.tag = idOfResource
+    }
+
+    /**
+     * Switch the location to on or off according to the current state.
+     * Accordingly update the icon.
+     */
+    @SuppressLint("MissingPermission")
+    private fun switchLocationOnOff() {
+        val locationIsOn = GoogleMapHelper.map!!.isMyLocationEnabled
+
+        if (locationIsOn) {
+            // Disable the location
+            GoogleMapHelper.map!!.isMyLocationEnabled = false
+            setLocationIcon(false)
+            useUserLocation = false
+        } else {
+            activateMyLocation()
+        }
+    }
+
+    /**
+     * Get and return the built-in "my location" button from map fragment.
+     * @return the view of the built-in location button
+     */
+    private fun getBuiltInLocationButton(): View {
+        val mapFragment =
+            childFragmentManager.findFragmentById(R.id.id_fragment_map) as SupportMapFragment?
+
+        // Magic : https://stackoverflow.com/questions/36785542/how-to-change-the-position-of-my-location-button-in-google-maps-using-android-st
```

Comment:
Is this magic?

Reply:
Yes ! It is to locate the built int location button from GoogleMap